Add documentation for Snaps lifecycle hooks (#1077)
MetaMask/metamask-docs: snaps/reference/permissions.md

}
```

### endowment:lifecycle-hooks

To run an action when the user installs or updates the Snap, a Snap must request the `endowment:lifecycle-hooks` permission.
This permission allows the Snap to export the 
[`onInstall`](../reference/exports.md#oninstall) and 
[`onUpdate`](../reference/exports.md#onupdate) 

> methods, which will be called by MetaMask after a successful install or update, respectively.

Reviewer: ```suggestion
methods, which MetaMask calls after a successful installation or update, respectively.
```
